modin/logging/logger_decorator.py

```diff
@@ -94,11 +113,11 @@ def decorator(obj: Any) -> Any:
                         )(attr_value)
 
                     setattr(obj, attr_name, wrapped)
-            return obj
+            return obj  # type: ignore [return-value]
         elif isinstance(obj, classmethod):
-            return classmethod(decorator(obj.__func__))
+            return classmethod(decorator(obj.__func__))  # type: ignore [return-value, arg-type]
         elif isinstance(obj, staticmethod):
-            return staticmethod(decorator(obj.__func__))
+            return staticmethod(decorator(obj.__func__))  # type: ignore [return-value, arg-type]
```

Comment:
why?

Reply:
I was unable to build `Fn` type to be compatible with `staticmethod`, `classmethod` and pure `type`.